odoo/addons/base/models/ir_actions.py

```diff
@@ -497,7 +497,10 @@ def run_action_object_create(self, action, eval_context=None):
 
         if action.link_field_id:
             record = self.env[action.model_id.model].browse(self._context.get('active_id'))
-            record.write({action.link_field_id.name: res.id})
+            if action.link_field_id.ttype == 'many2one':
+                record.write({action.link_field_id.name: res.id})
+            if action.link_field_id.ttype in ['one2many', 'many2many']:
+                record.write({action.link_field_id.name: [(4, res.id, True)]})
```

Comment:
The command "4" is commonly a pair, not a triple
```suggestion
                record.write({action.link_field_id.name: [(4, res.id)]})
```